src/engines/kotor/module.cpp

```diff
@@ -448,6 +448,37 @@ void Module::clickObject(Object *object) {
 	}
 }
 
+void Module::enterObject(Object *object) {
+	Placeable *placeable = ObjectContainer::toPlaceable(object);
+	if (placeable) {
+		_selected = placeable;
+		updateSelection();
+		return;
+	}
+}
+
+void Module::leaveObject(Object *UNUSED(object)) {
+	if (_selected) {
+		_selected = 0;
+		_ingame->hideSelectionCicle();
+	}
+}
+
+void Module::updateSelection() {
+	if (_selected) {
+		float x, y, z;
+		Placeable *placeable = ObjectContainer::toPlaceable(_selected);
+		if (placeable) {
+			placeable->getTooltipAnchor(x, y, z);
+
+			float sX, sY, sZ;
+			assert(GfxMan.project(x, y, z, sX, sY, sZ));
```

Comment:
`assert()`s shouldn't have any side effects. They're macros that will be disabled completely when compiling with -DNDEBUG.

I.e. you need to save the result of the function call in a variables and assert on that.

```
const bool result = GfxMan.project(x, y, z, sX, sY, sZ);
assert(result);
```